src/aria/widgets/form/CheckBox.js

```diff
@@ -124,21 +116,19 @@ module.exports = Aria.classDefinition({
             var cfg = this._cfg;
             var name = this._inputName ? ' name="' + this._inputName + '"' : "";
 
-            if (this._skinObj.simpleHTML) {
 
-                // tab index is needed for simple markup (not set on container span, otherwise double focus occurs)
-                var tabIndex = this._cfg.tabIndex;
-                tabIndex = !cfg.disabled ? 'tabindex="' + this._calculateTabIndex() + '" ' : "";
-
-                out.write(['<input style="display:inline-block"', cfg.disabled ? ' disabled' : '',
-                        this._isChecked() ? ' checked' : '', ' type="', cfg._inputType, '"', name, ' value="',
-                        cfg.value, '" ', tabIndex, this._waiAriaAttributes, '/>'].join(''));
-            } else {
+            if (!this._skinObj.simpleHTML) {
                 this._icon.writeMarkup(out);
-                out.write(['<input', Aria.testMode ? ' id="' + this._domId + '_input"' : '', ' style="display:none"',
-                        cfg.disabled ? ' disabled' : '', this._isChecked() ? ' checked' : '', ' type="',
-                        cfg._inputType, '"', name, ' value="', cfg.value, '"/>'].join(''));
             }
+            var tabIndex = this._cfg.tabIndex;
+            tabIndex = !cfg.disabled ? 'tabindex="' + this._calculateTabIndex() + '" ' : "";
+
+            out.write(['<input', cfg.disabled ? ' disabled' : '',
+                    this._skinObj.simpleHTML ? ' style="display:inline-block"' : ' class="xSROnly"',
+                    (Aria.testMode || cfg.waiAria) ? ' id="' + this._domId + '_input"' : '',
+                    this._isChecked() ? ' checked' : '', ' type="', cfg._inputType, '"', name, ' value="',
+                    cfg.value, '" ', tabIndex, this._waiAriaAttributes, '/>'].join(''));
```

Comment:
You should probably remove `this._waiAriaAttributes` as it is not used anywhere.